app/Models/Traits/Acceptable.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Models\Traits;
+
+use App\Models\Asset;
+use App\Models\CustomField;
+use App\Models\User;
+use Illuminate\Database\Eloquent\Builder;
+
+/**
+ * This trait allows models to have a callback after their checkout gets accepted or declined.
+ *
+ * @author Till Deeke <kontakt@tilldeeke.de>
+ */
+trait Acceptable {
+    /**
+     * Run after the checkout acceptance was accepted by the user
+     * 
+     * @param  User   $acceptedBy
+     * @param  string $signature
+     */
+    public function acceptedCheckout(User $acceptedBy, $signature) {}
+
+    /**
+     * Run after the checkout acceptance was declined by the user
+     * 
+     * @param  User   $acceptedBy
+     * @param  string $signature
+     */ 
+    public function declinedCheckout(User $declinedBy, $signature) {}
+}
```

Comment:
If neither method in this trait is implemented, is it really a trait?  I'm unsure of what it's currently doing.

Reply:
This basically acts as a no-op the checkout acceptance. After the user accepts/declines a checkout, these methods are called. If the model doesn't override them, they do nothing. 

The idea behind that was to allow clean methods in the CheckoutAcceptance model, while allowing the flexibility of running additional code after the user accepted/declined